### PR TITLE
Add recipe for txtorcon

### PR DESCRIPTION
**Pull request by tdido**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by tdido** [comment]
Hello @conda-forge/staged-recipes friends. A review would be much appreciated. Thanks!

File: recipes/txtorcon/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/txtorcon-{{ version }}.tar.gz
+  sha256: aebf0b9ec6c69a029f6b61fd534e785692e28fdcd2fd003ce3cc132b9393b7d6
+
+build:
+  number: 0
+  script: {{ PYTHON }} -m pip install . -vv
```
Comment (ocefpaf) [inline recipes/txtorcon/meta.yaml:15]: It looks like this recipe can be `noarch`, can you try that?

File: recipes/txtorcon/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    - twisted.plugins
+    - txtorcon
+  commands:
+    - pip check
+  requires:
+    - pip
+
+about:
+  home: https://github.com/meejah/txtorcon
+  summary: Twisted-based Tor controller client, with state-tracking and     configuration abstractions.     https://txtorcon.readthedocs.org     https://github.com/meejah/txtorcon
```
Comment (ocefpaf) [inline recipes/txtorcon/meta.yaml:42]: Do you mind fixing the bad spacing here?

**Comment by tdido** [comment]
@ocefpaf thank you for your help. It seems like the noarch build is working correctly.

**Pull request merged**